### DISCUSSION TITLE
Failed actions

### DISCUSSION THREAD

**Discussion by meryemberradauwo** [community/community#27912] [Actions] [answered]
Hello, 

I would like some help understanding what is happening with my repository "Rho". I am trying to create a submission for the Journal of Open Source Software and one of the files I had to add to my repository is a workflow to compile my files into a .pdf file. 
It appears that the workflow and all other actions that I have made so far have failed. I do not understand why and how to fix it.

I would greatly appreciate if someone would be willing to take a look. Thank you!

**Comment by airtower-luna** [comment]
Please include a link to the repository, workflow files, and runs so people can actually take a look. No-one can say if they can help without seeing the problem. :wink:

**Reply by meryemberradauwo** [reply]
Of course!
https://github.com/meryemberradauwo/Rho.git

**Comment by airtower-luna** [comment, marked as answer]
The workflow run says

> No jobs defined in `jobs`

but that looks like another case of a misleading error message. Your *jobs* section looks fine at a glance, but you have an indentation error on the first line:

```
 on: [push]

jobs:
```

There's a space in front of `on` that shouldn't be there. In YAML indentation defines structure, and *both* `on` and `jobs` must be at the top level.

I assume the YAML parser assumes the indentation of the first line must be the top level, and then the less-indented `jobs` is ignored. :sweat_smile:

**Reply by meryemberradauwo** [reply]
Thank you ! I was able to find the error description that you mention and fix it. I am now encountering another issue. It seems like the paper.bib file cannot be read. Would you able to take a look please ?

**Reply by airtower-luna** [reply]
Hm, the workflow syntax is fine now. The error you're seeing happens inside the [`openjournals/openjournals-draft-action` action](https://github.com/openjournals/openjournals-draft-action) you're using. What I can see without being familiar with that action:

* The error happens in metadata processing, there's a mention of a `nil` (missing/unset) value.
* Your `paper.md` looks like you're trying to use YAML metadata, but there's no end of document marker (`---` on a separate line, like line 1) between the metadata and the Markdown (before `# Summary`).

Try adding that marker. If that doesn't help you might want to write an issue in the action repository so the authors can take a look.

**Reply by meryemberradauwo** [reply]
Thank you so much! I added --- on a separate line under the paper.bib line in paper.md. 
Everythink works fine now! Thank you so much!

**Reply by airtower-luna** [reply]
Glad to hear it works! I'd appreciate if you'd mark the answer. :smiley_cat: 

And thanks for making me learn about that four-wire resistance measurement, that didn't come up in the electrical engineering classes I had in university. :smile_cat:
